PinClustering/PinClustering/Model/Managers/NetworkManager.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  NetworkManager.swift
+//  PinClustering
+//
+//  Created by Norbert Agoston on 06/09/16.
+//  Copyright © 2016 Norbert Agoston. All rights reserved.
+//
+
+import UIKit
+
+class NetworkManager: NSObject {
+    
+    static let sharedInstance = NetworkManager()
+    
+    func httpGet(url: NSURL, callback: (NSArray?, String?) -> Void) {
+        let configuration =  NSURLSessionConfiguration.defaultSessionConfiguration()
+        let session = NSURLSession(configuration: configuration)
+        let request = NSMutableURLRequest(URL: url)
+        
+        let task = session.dataTaskWithRequest(request) { (data, response, error) in
+            do {
+                if error != nil {
+                    callback(nil,error?.localizedDescription)
```

Comment:
Why don't you send the whole error in the completion block? By doing this you give the receiver more control over what to do with the error. The error could have more information than just the description.